### PR TITLE
Ft #157662524 - modify details of request

### PR DESCRIPTION
**Pull request by uchemukolo**
#### What does this PR do?
....

#### Description of Task to be completed?
API endpoints to:
- create a request
- modify details of a request
- Fetch all the requests of a logged in user
- Fetch a request that belongs to a logged in user

#### Any background context you want to provide?
....

#### What are the relevant Github Issues?
- conflicting files

#### Questions:

### CODE REVIEW

File: ui/css/main.css [hunk 1]
```diff
@@ -0,0 +1,343 @@
...
+    display:block;
+    margin-top:7px;
+    padding:0px 20px;
+    width:100%;
+}
+
+.navbar {
+    background-color: #ffff00;
+    display: flex;
+    height: 35px;
```
Comment (Tawakalt) [inline ui/css/main.css:181]: `height` should come after `font-weigh`.

File: ui/css/main.css [hunk 2]
```diff
@@ -0,0 +1,343 @@
...
+    color: #000000;
+    font-weight: bold;
+    text-align: left;
+}
+.container {
+    float: left;
+    margin: 6px auto;
+    text-align: left;
+    width: 30%;
+    padding: 10px;
```
Comment (Tawakalt) [inline ui/css/main.css:59]: `padding` should come before `text-align`

File: ui/css/main.css [hunk 3]
```diff
@@ -0,0 +1,343 @@
...
+    color: #000000;
+    font-weight: normal;
+    text-align: left;
+}
+.footer {
+    background: #222121;
+    color: #ffffff;
+    padding: 5px;
+    text-align: center;
+    margin-top: 8%;
```
Comment (Tawakalt) [inline ui/css/main.css:83]: `margin-top` should come before `padding`

File: ui/css/main.css [hunk 4]
```diff
@@ -0,0 +1,343 @@
...
+    margin-top: 6px;
+    margin: auto;
+    width: 75%;
+  }
+  input[type=submit] {
+    background-color: #ffff00;
+    border: none;
+    border-radius: 4px;
+    cursor: pointer;
+    color: #000000;
```
Comment (Tawakalt) [inline ui/css/main.css:121]: `color` should come before `cursor`

File: ui/css/main.css [hunk 5]
```diff
@@ -0,0 +1,343 @@
...
+    float: right;
+    margin:5px auto;
+    padding: 12px 20px;
+  }
+
+  input[type=text], select, textarea {
+    border: 1px solid #ccc;
+    border-radius: 4px;
+    resize: vertical;
```
Comment (Tawakalt) [inline ui/css/main.css:130]: `resize` should come after `padding`

File: ui/css/main.css [hunk 6]
```diff
@@ -0,0 +1,343 @@
...
+}
+.make_request {
+
+    float:left;
+    margin-top:7px;
+    margin-bottom: 7px;
+    padding:0px 20px 20px 0px;
+    text-align:center;
+    width:30%;
+    background-color: #f3f1f1
```
Comment (Tawakalt) [inline ui/css/main.css:143]: `background-color` should come first.

File: ui/css/main.css [hunk 7]
```diff
@@ -0,0 +1,343 @@
...
+}
+
+.main1 {
+    float:right;
+    margin-top:7px;
+    margin-bottom: 7px;
+    padding:0px 20px 20px 0px;
+    text-align:center;
+    width:20%;
+    background-color: #f3f1f1
```
Comment (Tawakalt) [inline ui/css/main.css:158]: `background-color` should come first.

File: ui/css/main.css [hunk 8]
```diff
@@ -0,0 +1,343 @@
...
+}
+.menu {
+    float:left;
+    margin-top:7px;
+    margin-bottom: 7px;
+    margin-left: 4px;
+    margin-right: 4px;
+    padding:0px 20px 20px 0px;
+    width: 69%;
+    background-color: #f3f1f1
```
Comment (Tawakalt) [inline ui/css/main.css:168]: `background-color` should come first.

File: ui/css/main.css [hunk 9]
```diff
@@ -0,0 +1,343 @@
...
+    margin: 0 auto;
+}
+
+.pagination a {
+    color: black;
+    float: left;
+    padding: 8px 16px;
+    text-decoration: none;
+    transition: background-color .3s;
+    border: 1px solid #ddd;
```
Comment (Tawakalt) [inline ui/css/main.css:221]: `border` should come first.

File: ui/css/main.css [hunk 10]
```diff
@@ -0,0 +1,343 @@
...
+}
+.row1 {
+    font-weight: bold;
+    text-decoration: none;
+    text-align: center;
+}
+.request_list  {
+    float: right;
+    margin-top:7px;
+    margin-bottom: 7px;
```
Comment (Tawakalt) [inline ui/css/main.css:245]: `margin-bottom` should come before `margin-top`.

File: ui/css/main.css [hunk 11]
```diff
@@ -0,0 +1,343 @@
...
+    padding:0px 20px 20px 0px;
+    text-align:center;
+    width:69%;
+}
+.request_list  {
+    float: left;
+    margin: auto;
+    margin-top:7px;
+    margin-bottom: 7px;
```
Comment (Tawakalt) [inline ui/css/main.css:254]: `margin-bottom` should come before `margin-top`.

File: ui/css/main.css [hunk 12]
```diff
@@ -0,0 +1,343 @@
...
+    padding:0px 100px 20px 100px;
+    text-align:center;
+    width:100%;
+}
+.request_id {
+    font-weight: bold;
+    text-decoration: none;
+    text-align: center;
```
Comment (Tawakalt) [inline ui/css/main.css:262]: `text-align` should come before `text-decoration`.

File: ui/css/main.css [hunk 13]
```diff
@@ -0,0 +1,343 @@
...
+}
+.request_id:hover {
+    color: #ff0000;
+}
+
+#search {
+    background-image: url('../image/search.png');
+    background-position: 10px 10px;
+    background-repeat: no-repeat;
+    width: 100%;
```
Comment (Tawakalt) [inline ui/css/main.css:272]: `width` should come last.

File: ui/css/main.css [hunk 14]
```diff
@@ -0,0 +1,343 @@
...
+    font-size: 16px;
+    padding: 12px 20px 12px 40px;
+    border: 1px solid #ddd;
```
Comment (Tawakalt) [inline ui/css/main.css:275]: `border` should come after `background-repeat`.

File: ui/css/main.css [hunk 15]
```diff
@@ -0,0 +1,343 @@
...
+    text-align: center;
+}
+.user_info {
+    font-weight: bold;
+    text-align: center;
+}
+
+.welcome {
+    font-size: 100px;
+    font-family: 'Franklin Gothic Medium', 'Arial Narrow', Arial, sans-serif;
```
Comment (Tawakalt) [inline ui/css/main.css:325]: `font-family` before `font-size`.

File: ui/css/signup.css [hunk 16]
```diff
@@ -4,59 +4,56 @@
...
+    padding: 0px;
+    text-align: center;
+}
+
+.container {
+    border-radius: 10px;
+    margin: auto;
+    padding: 10px 0px 10px 0px;
+    width: 60%;
+    background-color: #f3f1f1;
```
Comment (Tawakalt) [inline ui/css/signup.css:17]: `background-color` first.

File: ui/css/signup.css [hunk 17]
```diff
@@ -4,59 +4,56 @@
...
 
 input[type=submit] {
-    background-color: yellow;
-    color: black;
-    padding: 12px 20px;
+    background-color: #ffff00;
     border: none;
     border-radius: 4px;
     cursor: pointer;
+    color: #000000;
```
Comment (Tawakalt) [inline ui/css/signup.css:36]: `color` before `cursor`

File: ui/css/signup.css [hunk 18]
```diff
@@ -4,59 +4,56 @@
...
-
-
+    background-color: #ffffff;
 }
 
 .input_field {
     float: left;
-    width: 75%;
     margin-top: 6px;
     margin: auto;
```
Comment (Tawakalt) [inline ui/css/signup.css:49]: `margin` before `margin-top`

File: ui/requestdetails.html [hunk 19]
```diff
@@ -0,0 +1,40 @@
+<!DOCTYPE html>
+<html>
+<head>
+    <title>Request Details</title>
+    <meta charset="UTF-8">
+    <meta name="viewport" content="width=device-width, initial-scale=1">
+    <link rel="stylesheet" href="css/signup.css">
+    <link rel="stylesheet" href="css/main.css">
```
Comment (Tawakalt) [inline ui/requestdetails.html:8]: `<link rel="stylesheet" href="css/main.css">` before `<link rel="stylesheet" href="css/signup.css">`

File: ui/requestdetails_admin.html [hunk 20]
```diff
@@ -0,0 +1,42 @@
+<!DOCTYPE html>
+<html>
+<head>
+    <title>Request Details</title>
+    <meta charset="UTF-8">
+    <meta name="viewport" content="width=device-width, initial-scale=1">
+    <link rel="stylesheet" href="css/signup.css">
+    <link rel="stylesheet" href="css/main.css">
```
Comment (Tawakalt) [inline ui/requestdetails_admin.html:8]: `<link rel="stylesheet" href="css/main.css">` before `<link rel="stylesheet" href="css/signup.css">`

File: ui/resolve.html [hunk 21]
```diff
@@ -0,0 +1,42 @@
+<!DOCTYPE html>
+<html>
+<head>
+    <title>Request Details</title>
+    <meta charset="UTF-8">
+    <meta name="viewport" content="width=device-width, initial-scale=1">
+    <link rel="stylesheet" href="css/signup.css">
+    <link rel="stylesheet" href="css/main.css">
```
Comment (Tawakalt) [inline ui/resolve.html:8]: `<link rel="stylesheet" href="css/main.css">` before `<link rel="stylesheet" href="css/signup.css">`

File: ui/signin.html [hunk 22]
```diff
@@ -1,31 +1,28 @@
 <!DOCTYPE html>
 <html>
 <head>
-    <title>Signup</title>
+    <title>Sign In</title>
     <meta charset="UTF-8">
     <meta name="viewport" content="width=device-width, initial-scale=1">
     <link rel="stylesheet" href="css/signup.css">
     <link rel="stylesheet" href="css/main.css">
```
Comment (Tawakalt) [inline ui/signin.html:8]: `<link rel="stylesheet" href="css/main.css">` before `<link rel="stylesheet" href="css/signup.css">`

File: server/app.js [hunk 23]
```diff
@@ -0,0 +1,24 @@
+import express from 'express';
+import logger from 'morgan';
+import bodyParser from 'body-parser';
```
Comment (Tawakalt) [inline server/app.js:3]: `import bodyParser from 'body-parser'` should come first

File: server/db/request.js [hunk 24]
```diff
@@ -0,0 +1,21 @@
...
+	{
+		userId: 1,
+		requestId: 1,
+		title:'Faulty Laptop',
+		category: 'Repair',
+    description:' My laptop is not coming up. Yesterday everything was working fine but this morning I noticed the laptop was very hot and not coming up',
+		urgencyLevel:'High',
+		date: '15/05/2018'
+	}
+	// {
```
Comment (Tawakalt) [inline server/db/request.js:11]: Remove all commented-out code

File: ui/css/main.css [hunk 25]
```diff
@@ -0,0 +1,343 @@
...
+
+body {
+    font-family: Arial, Helvetica, sans-serif;
+    margin: 0;
+
+}
+.button {
+    border: 2px solid #ffff00;
+    border-radius: 5px;
+    background-color: #ffff00;
```
Comment (Tawakalt) [inline ui/css/main.css:13]: `background-color` should come first

File: ui/css/main.css [hunk 26]
```diff
@@ -0,0 +1,343 @@
...
+    color: #000000;
+    font-size: 30px;
+    margin-right: 20px;
+    padding: 10px 25px 10px 25px;
+    text-decoration: none;
+}
+.button1 {
+    border: 2px solid #ffff00;
+    border-radius: 5px;
+    background-color: #ffff00;
```
Comment (Tawakalt) [inline ui/css/main.css:23]: `background-color` should come first

File: ui/css/main.css [hunk 27]
```diff
@@ -0,0 +1,343 @@
...
+    color: #000000;
+    font-size: 30px;
+    padding: 10px 25px 10px 25px;
+    text-decoration: none;
+}
+.button_d {
+    border: 2px solid #ffff00;
+    border-radius: 5px;
+    background-color: #ffff00;
```
Comment (Tawakalt) [inline ui/css/main.css:32]: `background-color` should come first

File: ui/css/main.css [hunk 28]
```diff
@@ -0,0 +1,343 @@
...
+    color: #000000;
+    font-size: 15px;
+    margin-right: 20px;
+    padding: 10px 25px 10px 25px;
+    text-decoration: none;
+}
+.button_d1 {
+    border: 2px solid #ffff00;
+    border-radius: 5px;
+    background-color: #ffff00;
```
Comment (Tawakalt) [inline ui/css/main.css:42]: `background-color` should come first

File: ui/css/main.css (hunk 2 above)
Comment (Tawakalt) [inline ui/css/main.css]: `padding` before `text-align`

File: ui/css/main.css [hunk 29]
```diff
@@ -0,0 +1,343 @@
...
+.main {
+    background-color: #ffffff;
+    flex:100%;
+    padding: 20px;
+}
+
+.main1 {
+    float:right;
+    margin-top:7px;
+    margin-bottom: 7px;
```
Comment (Tawakalt) [inline ui/css/main.css:154]: `margin-bottom` before `margin-top`

File: ui/css/main.css [hunk 30]
```diff
@@ -0,0 +1,343 @@
...
+    padding:0px 20px 20px 0px;
+    text-align:center;
+    width:20%;
+    background-color: #f3f1f1
+}
+.menu {
+    float:left;
+    margin-top:7px;
```
Comment (Tawakalt) [inline ui/css/main.css:162]: use `margin: 7px 4px ` instead of listing all four margin-types

File: ui/requestdetails.html (hunk 19 above)
Comment (uchemukolo) [reply ui/requestdetails.html:8]: ok

**Pull request closed without merging**